File: Latihan B/PHP/Memory.php
<?php

//kelas yang digunakan untuk mengimplementasikan kelas Memory
class Memory extends Hardware{

	//atribut private
	private $frequency;
	private $memorySize;
	private $supportsCuda;

	//konstruktor
	public function __construct(){
		
	}

	//getter dan setter untuk atribut frequency
	public function setFrequency($frequency){
		$this->frequency = $frequency;
	}

	public function getFrequency(){
		return $this->frequency;
	}

	//getter dan setter untuk atribut memorySize
	public function setMemorySize($memorySize){
		$this->memorySize = $memorySize;
	}

	public function getMemorySize(){
		return $this->memorySize;
	}

	//getter dan setter untuk atribut supportsCuda
	public function setSupportsCuda($supportsCuda){
		$this->supportsCuda = $supportsCuda;
	}

	public function getSupportsCuda(){
		return $this->supportsCuda;
	}

	//destruktor
	public function __destruct(){

	}

}

?>

File: Latihan B/C++/Memory.cpp
//kelas yang digunakan untuk mengimplementasikan kelas Memory
class Memory : public Hardware{

	private:

		//atribut private
		int frequency;
		int memorySize;
		string supportsCuda;

	public:

		//konstruktor
		Memory(int frequency, int memorySize, string supportsCuda){
			this->frequency = frequency;
			this->memorySize = memorySize;
			this->supportsCuda = supportsCuda;
		}

		Memory(){
			this->frequency = 0;
			this->memorySize = 0;
			this->supportsCuda = "";
		}

		//getter dan setter frequency
		void setFrequency(int frequency){
			this->frequency = frequency;
		}

		int getFrequency(){
			return this->frequency;
		}

		//getter dan setter memorySize
		void setMemorySize(int memorySize){
			this->memorySize = memorySize;
		}

		int getMemorySize(){
			return this->memorySize;
		}

		//getter dan setter supportsCuda
		void setSupportsCuda(string supportsCuda){
			this->supportsCuda = supportsCuda;
		}

		string getSupportsCuda(){
			return this->supportsCuda;
		}

		//destruktor
		~Memory(){

		}

};